File: customer-service/src/main/java/com/wks/servicemarketplace/customerservice/config/ClientCredentialsTokenSupplierFactory.java
package com.wks.servicemarketplace.customerservice.config;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.wks.servicemarketplace.authservice.api.ClientCredentialsRequest;
import com.wks.servicemarketplace.authservice.api.ClientCredentialsTokenSupplier;
import org.glassfish.hk2.api.Factory;

import javax.inject.Inject;
import java.util.Collections;

public class ClientCredentialsTokenSupplierFactory implements Factory<ClientCredentialsTokenSupplier> {

    private ClientCredentialsTokenSupplier clientCredentialsTokenSupplier;

    @Inject
    public ClientCredentialsTokenSupplierFactory(ApplicationParameters applicationParameters, ObjectMapper objectMapper){
        clientCredentialsTokenSupplier = new ClientCredentialsTokenSupplier(
                new ClientCredentialsRequest(applicationParameters.getClientId(), applicationParameters.getClientSecret()),
                applicationParameters.getAuthServiceBaseUrl(),
                objectMapper
        );
    }

    @Override
    public ClientCredentialsTokenSupplier provide() {
        return clientCredentialsTokenSupplier;
    }

    @Override
    public void dispose(ClientCredentialsTokenSupplier instance) {

    }
}

File: auth-service/src/main/kotlin/com/wks/servicemarketplace/authservice/config/ClientCredentialsTokenSupplierFactory.kt
package com.wks.servicemarketplace.authservice.config

import com.fasterxml.jackson.databind.ObjectMapper
import com.wks.servicemarketplace.authservice.api.ClientCredentialsRequest
import com.wks.servicemarketplace.authservice.api.ClientCredentialsTokenSupplier
import org.glassfish.hk2.api.Factory
import javax.inject.Inject

class ClientCredentialsTokenSupplierFactory @Inject constructor(applicationParameters: ApplicationParameters, objectMapper: ObjectMapper) : Factory<ClientCredentialsTokenSupplier> {

    private val clientCredentialsTokenSupplier = ClientCredentialsTokenSupplier(
            ClientCredentialsRequest(applicationParameters.clientId, applicationParameters.clientSecret),
            applicationParameters.authServiceBaseUrl,
            objectMapper
    )

    override fun provide() = clientCredentialsTokenSupplier

    override fun dispose(instance: ClientCredentialsTokenSupplier?) {
    }

}